src/Xamarin.Android.Build.Tasks/Utilities/MavenExtensions.cs

```diff
@@ -54,47 +56,76 @@ public static bool TryParseArtifactWithVersion (string id, string version, TaskL
 		return true;
 	}
 
-	public static bool TryParseJavaArtifactAndJavaVersion (this ITaskItem task, string type, TaskLoggingHelper log, [NotNullWhen (true)] out Artifact? artifact, out bool attributesSpecified)
+	public static bool TryParseArtifacts (string id, TaskLoggingHelper log, out List<Artifact> artifacts)
 	{
-		artifact = null;
-		var item_name = task.ItemSpec;
+		artifacts = new List<Artifact> ();
+		var result = true;
 
-		// Convert "../../src/blah/Blah.csproj" to "Blah.csproj"
-		if (type == "ProjectReference")
-			item_name = Path.GetFileName (item_name);
+		var arts = id.Split (artifacts_separators, StringSplitOptions.RemoveEmptyEntries);
 
-		var has_artifact = task.HasMetadata ("JavaArtifact");
-		var has_version = task.HasMetadata ("JavaVersion");
+		foreach (var art in arts) {
+
+			if (Artifact.TryParse (art, out var a)) {
+				artifacts.Add (a);
+				continue;
+			}
+
+			log.LogCodedError ("XA4249", Properties.Resources.XA4249, art);
+			result = false;
+		}
 
-		// Lets callers know if user attempted to specify JavaArtifact or JavaVersion, even if they did it incorrectly
-		attributesSpecified = has_artifact || has_version;
+		return result;
+	}
 
-		if (has_artifact && !has_version) {
-			log.LogCodedError ("XA4243", Properties.Resources.XA4243, "JavaVersion", "JavaArtifact", type, item_name);
+	public static bool TryParseJavaArtifact (this ITaskItem task, string type, TaskLoggingHelper log, [NotNullWhen (true)]out Artifact? artifact, out bool attributesSpecified)
+	{
+		var result = TryParseJavaArtifacts (task, type, log, out var artifacts, out attributesSpecified);
+
+		if (!result) {
+			artifact = null;
 			return false;
 		}
 
-		if (!has_artifact && has_version) {
-			log.LogCodedError ("XA4243", Properties.Resources.XA4243, "JavaArtifact", "JavaVersion", type, item_name);
+		// TODO: Need a new message saying that only one JavaArtifact is allowed
+		if (artifacts.Count > 1) {
+			log.LogCodedError ("XA4243", Properties.Resources.XA4243, "JavaArtifact", type, task.ItemSpec);
+			artifact = null;
 			return false;
 		}
 
-		if (has_artifact && has_version) {
+		artifact = artifacts.FirstOrDefault ();
+
+		return artifact is not null;
+	}
+
+	public static bool TryParseJavaArtifacts (this ITaskItem task, string type, TaskLoggingHelper log, out List<Artifact> artifacts, out bool attributesSpecified)
+	{
+		artifacts = new List<Artifact> ();
+		var item_name = task.ItemSpec;
+
+		// Convert "../../src/blah/Blah.csproj" to "Blah.csproj"
+		if (type == "ProjectReference")
+			item_name = Path.GetFileName (item_name);
```

Comment:
Do we actually want to do this?  What if the same project name is used multiples but for different paths: `Lib1/Helper.csproj`, `Lib2/Helper.csproj`, …?